.github/workflows/build-and-publish.yaml

```diff
@@ -43,6 +43,8 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           python -m pip install tox
+          python -m pip install poetry --user
```

Comment:
why is there a need to install this to `--user`? If you can install peotry the same way as tox, that this obliviates the need for the `PATH` setting below.

Reply:
There quite a few instances of these in other worflows as well.

Reply:
tox always creates it's own isolated venv.
Agree it would probably work without `--user`

Reply:
Good catch, `tox` and `poetry` should go in the same virtual-environment and that environment should not be the one that we install `fms-hf-tuning` in. Because `tox`  creates a new virtual environment for the `environments` it processes (e.g. `py`, `fmt, etc) poetry and tox will be outside the tested virtual-environment.

We don't need to run tox "inside" one of the virtual environments that tox creates but we do need to run `poetry`. Here's how I chose to do that.

First, I installed poetry under the user pip install directory (which by default for linux is `~/.local/`. Then I updated the file that [$GITHUB_ENV](https://docs.github.com/en/actions/learn-github-actions/variables#passing-values-between-steps-and-jobs-in-a-workflow) points to so that in subsequent steps the `$PATH` environment variable contains the path `~/.local/bin`. 

This ensures that the `poetry` commands running inside `tox` can use the `poetry` executable. Alternatively, we can create a new virtual environment under a directory of our choosing e.g. `/tmp/isolated` in there, we install just poetry and then update `$GITHUB_ENV` so that `$PATH` includes `/tmp/isolated/bin`.